CDCgov/prime-reportstream
# closes #2797 run gitleaks locally if installed

This PR is part of the effort to reduce docker requirements for githooks. This PR addresses gitleaks specifically and will instead run gitleaks locally if installed.

Test Steps:
1. Run `./.environment/gitleaks/run-gitleaks.sh`
   * If gitleaks is not installed locally, gitleaks will still run in a docker container
2. [Install gitleaks](https://github.com/zricethezav/gitleaks#from-bin)
3. gitleaks should now run locally.

## Changes
1. Modified `run-gitleaks.sh` to run gitleaks locally if installed

## Checklist

### Testing

## Fixes
- #2797 
- docker has already been removed from [ktlint](https://github.com/CDCgov/prime-reportstream/commit/a3b488553b9078104b672b0bf861f25f77ffe4d1), so this is the last remaining docker githook
